python_ta/checkers/bad_loop_indexing_checker.py

```diff
@@ -0,0 +1,109 @@
+"""checker for bad indexing in a loop.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from typing import List, Optional
+from astroid.node_classes import NodeNG
+
+
+class BadLoopIndexingChecker(BaseChecker):
```

Comment:
The terms "good" and "bad" are vague and should generally be avoided when naming things. Here, you should rename this checker (and the corresponding files/methods/etc.) in the following ways:

1. Replace references to "bad loop indexing" with "unnecessary indexing".
2. Replace references to "bad usage" with "redundant index use". (The term "redundant" here means that the index name could have been replaced with a simpler reference to the list element. This should be added to the docstring of a function that checks whether the use is 'redundant' or not.)